frontend/components/Editor.js

```diff
@@ -665,6 +671,16 @@ patch: ${JSON.stringify(
         const on_establish_connection = async (client) => {
             // nasty
             Object.assign(this.client, client)
+            try {
+                const { default: environment } = await import(this.client.session_options.server.injected_javascript_data_url)
+                const { custom_editor_header_component } = environment(client, html, useEffect, useState, useMemo)
```

Comment:
To make this more future-proof, it could be:
```js
environment({
	client,
	editor: this,
	imports: {  preact }
})
```

Passing in `this` allows the extension to get anything it wants, like `editor.state.notebook_id`

And we add this at the top:
```js
import * as preact from "../imports/Preact.js"
```